api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,222 @@
+	//    allocatable:
+	//      cpu:
+	//        min: "16"
+	//        max: "64"
+	//      memory:
+	//        min: "128Gi"
+	//        max: "512Gi"
+	//      accelerators:
+	//        nvidia.com/gpu:
+	//          min: "2"
+	//	        max: "8"
+	//  acceleratorSelector:
+	//    type: "nvidia.com/gpu"
+	//    memory: "48Gi"
+	//    count: 4
+	// +required
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// PerAcceleratorCount defines pre-configuration quantities for each accelerator type
+	PerAcceleratorCount []PerAcceleratorCount `json:"perAcceleratorCount"`
+}
+
+// EnhancedNodeSelector defines node selector with label selector and resource requirements.
+type EnhancedNodeSelector struct {
+	// LabelSelector defines the label selector for a node.
+	// +required
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+	// ResourceRequirements defines the resource requirements for a node.
+	// +optional
+	ResourceRequirements *ResourceRequirements `json:"resourceRequirements,omitempty"`
+	// AcceleratorSelector defines accelerator-specific selection criteria
+	AcceleratorSelector *AcceleratorSelector `json:"acceleratorSelector,omitempty"`
+}
+
+// PerAcceleratorCount defines configuration for specific accelerators
+type PerAcceleratorCount struct {
+	// AcceleratorType specifies accelerator type (e.g., nvidia.com/gpu)
+	AcceleratorType string `json:"acceleratorType,omitempty"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node per accelerator.
+	// If two different counts are specified for the same (Node, Accelerator, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, Accelerator, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +required
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirements defines resource requirements for a node.
+type ResourceRequirements struct {
+	// Allocatable defines the allocatable resources for a node with min/max ranges.
+	// +required
+	Allocatable ResourceRanges `json:"allocatable,omitempty"`
+}
+
+// ResourceRanges defines min/max ranges for various resources of a Node.
+type ResourceRanges struct {
+	// CPU defines the CPU resource range requirement.
+	// +optional
+	CPU ResourceRange `json:"cpu,omitempty"`
+
+	// Memory defines the memory resource range requirement.
+	// +optional
+	Memory ResourceRange `json:"memory,omitempty"`
+
+	// Accelerators defines the GPU resource range requirements keyed by GPU type.
+	// +optional
+	Accelerators map[string]ResourceRange `json:"accelerators,omitempty"`
+}
+
+// ResourceRange defines a range with minimum and maximum quantity values.
+type ResourceRange struct {
+	// Min specifies the minimum quantity required.
+	// +optional
+	Min resource.Quantity `json:"min,omitempty"`
```

Comment:
Since this is optional, I think you want a level of pointer indirection here. The `,omitempty` is meaningless here because the marshaled form of `Resource.Quantity` has no way of saying "empty".

Reply:
sure, resolved on latest version